### PR TITLE
📖 Fix a bunch of broken links in docs

### LINKED ISSUE #1964: bug: docs/content/direct/README.md has broken link to user-guide.md
### Describe the bug

https://github.com/kubestellar/kubestellar/blob/main/docs/content/direct/README.md refers to a file that does not exist, `user-guide.md`. PR #1962 renamed that file but did not update the reference here. The re-org done in that PR was exclusively focused on the view available through the website and ignored docs/content/direct/README.md.

### Steps To Reproduce

See above.

### Expected Behavior

No docs broken. The docs/content/direct/ content is viewable both through the website and through GitHub's direct view.

### Additional Context

_No response_

### LINKED ISSUE #1984: bug: links from docs to non-doc repo content fail on website
### Describe the bug

There are many relative links from markdown files in `docs/content/direct/` to non-markdown files or directories elsewhere in the same repo (or, in some cases, org). These work OK when the referring markdown file is viewed directly at GitHub, but in the rendered GitHub pages these links are broken.

An example is the reference to the `transport.go` file at https://github.com/kubestellar/kubestellar/blob/v0.21.1/docs/content/direct/architecture.md?plain=1#L449 , which is broken when viewed at https://docs.kubestellar.io/release-0.21.1/direct/architecture/#transport-controller .

### Steps To Reproduce

See above.

### Expected Behavior

Such links as we have work properly both when viewed on the website and when viewed directly at GitHub.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR fixes a lot of broken links in documents.
This PR also brings the document about the documentation system into the website.

This PR replaces a bunch of links to non-document files with non-links, because there is no good way to write a link that works both when the doc is viewed on the website and when the doc source is viewed directly through GitHub. In fact, there is no _good_ way to write a link to non-doc repo content that works properly when viewed at the website at all. There almost is. How to make a link that does not get broken by later re-orgs of the files? For the website, the link can use `{{ config.ks_branch }}` (see https://github.com/kubestellar/kubestellar/blob/v0.21.1/docs/README.md#global-variables), that will get replaced by the release identifier IF we follow an enabling discipline with release branches. The enabling discipline has to get `ks_branch: X.Y.Z` in `docs/mkdocs.yml` in branch `release-X.Y.Z`. And keep `ks_branch: main` in `docs/mkdocs.yml` in branch `main`. This forces release branches to have positively different content than `main` ever had. This is not the way that we are rolling right now.

## Related issue(s)

Fixes #1964 
Fixes #1984

### CODE REVIEW

File: docs/content/direct/contributor.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ make test
 
 ## Integration testing
 
-There are currently two integration tests. Contributors can run them. There is also [a GitHub workflow](../../../.github/workflows/pr-test-integration.yml) that runs these tests.
+There are currently two integration tests. Contributors can run them. There is also a GitHub workflow (in `.github/workflows/pr-test-integration.yml`) that runs these tests.
```
Comment (waltforme) [inline docs/content/direct/contributor.md:18]: I think the formal/complete name is GitHub 'Actions' workflow.

File: docs/content/direct/examples.md [hunk 2]
```diff
@@ -1,10 +1,8 @@
 # KubeStellar Usage Examples
 
-This document shows some simple examples of using the release that contains this version of this document. See also [the start of a user guide](user-guide.md) for considerations for different scenarios. For historical examples, see [our blog posts](https://medium.com/@kubestellar/list/predefined:e785a0675051:READING_LIST).
+This document shows some simple examples of using the release that contains this version of this document. See also [the doc host using an existing hosting cluster and multi-machine scenarios](hosting-cluster.md) for considerations for different scenarios. For historical examples, see [our blog posts](https://medium.com/@kubestellar/list/predefined:e785a0675051:READING_LIST).
```
Comment (waltforme) [inline docs/content/direct/examples.md:3]: Is the added word 'host' necessary?

File: docs/content/direct/packaging.md [hunk 3]
```diff
@@ -201,13 +201,13 @@ platform. This is used in local testing.
...
 The source for the Helm chart is in
-[chart](../../../chart). `make chart` (re)derives
+the `chart/` directory. `make chart` (re)derives
 it from local sources. This is not included in `make all-generated`.
 
 This chart creates (among other things) a `Deployment` object that runs a container from the [KubeStellar controller-manager container image](#kubestellar-controller-manager-container-image).
 
 The chart is published at the OCI repository
-`ghcr.io/kubestellar/kubestellar/kubestellar-operator-chart`. A [GitHub workflow](../../../.github/workflows/goreleaser.yml) specializes and publishes this chart as part of [the release process](release.md).
+`ghcr.io/kubestellar/kubestellar/kubestellar-operator-chart`. A GitHub workflow (in `.github/workflows/goreleaser.yml`) specializes and publishes this chart as part of [the release process](release.md).
```
Comment (waltforme) [inline docs/content/direct/packaging.md:210]: Same GitHub 'Actions' workflow.

File: docs/content/direct/contributor.md (hunk 1 above)
Comment (MikeSpreitzer) [reply docs/content/direct/contributor.md]: Fixed, thanks.

File: docs/content/direct/packaging.md (hunk 3 above)
Comment (MikeSpreitzer) [reply docs/content/direct/packaging.md]: fixed.

**Pull request merged**